script/subtitles/imscsubtitles.js

```diff
@@ -84,24 +146,43 @@ define('bigscreenplayer/subtitles/imscsubtitles',
       }
 
       function update (currentTime) {
-        var subtitlesIndex = nextSubtitleIndex(currentTime);
-        var generateAndRender = subtitlesIndex !== previousSubtitlesIndex;
+        var segment = getSegmentToRender(currentTime);
 
-        if (generateAndRender) {
-          render(currentTime);
-          previousSubtitlesIndex = subtitlesIndex;
+        if (segment) {
+          DebugTool.keyValue({key: 'segmentToRender', value: 'true'});
+          render(currentTime, segment.xml);
+        } else {
+          DebugTool.keyValue({key: 'segmentToRender', value: 'false'});
         }
       }
 
-      function render (currentTime) {
+      function getSegmentToRender (currentTime) {
```

Comment:
We made a decision to use the term segment instead of fragment.

Reply:
Is there an industry standard naming convention? I know in the BBC guidelines we tend to use fragment. What does DASH or the TextTrack API use for example?

It could be useful to be able to distinguish between media segments and subtitle fragments 🤷

Reply:
No idea. We originally had segment/fragment everywhere and it was more confusing, fragmented / segmented mean the same thing to me. Since we fetch captions using the same method as media it made more sense to use segment.

Reply:
As long as we converge on one or the other and we're not mixing within the same context (ie. always referenced as segment in this code) then I guess it's fine for now. We can always agree on better naming another time. 

Unless there are strong thoughts otherwise?